sqlglot/parser.py

```diff
@@ -367,6 +368,7 @@ class Parser(metaclass=_Parser):
         TokenType.USERDEFINED,
         TokenType.MONEY,
         TokenType.SMALLMONEY,
+        TokenType.RANGE,
```

Comment:
We don't need this here, `NESTED_TYPE_TOKENS` is expanded within this set below.